test/test_core.py

```diff
@@ -2636,6 +2636,13 @@ def test_pthread_thread_local_storage(self):
       self.set_setting('INITIAL_MEMORY', '300mb')
     self.do_run_in_out_file_test('pthread/test_pthread_thread_local_storage.cpp')
 
+  @node_pthreads
+  @also_with_minimal_runtime
+  def test_pthread_c_thread_local(self):
+    if self.get_setting('MINIMAL_RUNTIME') and is_sanitizing(self.cflags):
+      self.skipTest('MINIMAL_RUNTIME + threads + asan does not work')
+    self.do_run_in_out_file_test('pthread/test_pthread_c_thread_local.c')
```

Comment:
Surely we already have tests that cover this?

Reply:
It looks like `test/pthread/test_pthread_tls.cpp` is probably what we want.  It happens to use C++ `thread_local` instead of the C's `_Thread_local` but I think it should the same thing from the POV of this PR.